core/impl/src/main/java/com/blazebit/persistence/impl/JoinVisitor.java

```diff
@@ -74,6 +76,24 @@ private void visit(PathExpression expression, boolean idRemovable) {
             aliasedExpression.accept(this);
         } else {
             joinManager.implicitJoin(expression, joinWithObjectLeafAllowed, null, fromClause, false, false, joinRequired, idRemovable);
+            if (parentVisitor != null) {
+                JoinNode baseNode = (JoinNode) expression.getBaseNode();
+                AliasManager aliasOwner = baseNode.getAliasInfo().getAliasOwner();
+                if (aliasOwner != joinManager.getAliasManager()) {
+                    parentVisitor.addClauseDependencies(baseNode, aliasOwner);
+                }
+            }
+        }
+    }
+
+    void addClauseDependencies(JoinNode node, AliasManager aliasOwner) {
```

Comment:
Why is this method not private?

Reply:
Otherwise I couldn't call the method on a different instance.

Reply:
You are only calling it on a different instsance of the same type, so it can be private.